client/pubsub.go

```diff
@@ -8,17 +8,21 @@ import (
 )
 
 // PublishEvent pubishes data onto specific pubsub topic.
-func (c *GRPCClient) PublishEvent(ctx context.Context, topic string, in []byte) error {
+func (c *GRPCClient) PublishEvent(ctx context.Context, component, topic string, in []byte) error {
 	if topic == "" {
```

Comment:
Might want to make this the same as `AddTopicEventHandler` and be consistent with the validation messages:

```
if topic == "" {
		return fmt.Errorf("topic name required")
	}
	if component == "" {
		return fmt.Errorf("component name required")
	}
```

Reply:
Validation messages are now consistent